homeassistant/components/sensor/rest.py

```diff
@@ -40,6 +42,7 @@
     vol.Optional(CONF_USERNAME): cv.string,
     vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
     vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+    vol.Optional(CONF_JSON_ATTRS): cv.ensure_list_csv,
```

Comment:
According to your docs the default is an empty list (-> `default=[]`). To be in sync with other platform we should only accept a list and not a comma-separated string ->`vol.Optional(CONF_JSON_ATTRS): vol.All(cv.ensure_list, [cv.string]),`

Reply:
I've added the default (although the existing code treats the a value of `None` the same as an empty list).

Regarding the use of `cv.ensure_list_csv`, is this function deprecated? The function seems to be there explicitly to allow users to enter lists either as a YAML list or as a CVS list, which is very helpful to users and allows them to choose the representation that they find most readable and compact. The function gets used by calendar/todoist.py, insteon_plm, sensor/mvglive.py and switch/rpi_rf.py and serves a very useful purpose. If the intention is to do way with this useful capability then I can remove it from here but if the function is going to stay around is there really a compelling reason not to use it?

Reply:
No, it's not deprecated and was introduced for the customizing part. I was wrong...it works for all possible ways:

```yaml
    json_attributes:
      - "time"
      - "date"
      - "milliseconds_since_epoch"
    json_attributes: ["time", "date", "milliseconds_since_epoch"]
    json_attributes: time, date, milliseconds_since_epoch
    json_attributes: time,date,milliseconds_since_epoch
    json_attributes: "time", "date", "milliseconds_since_epoch"
```
But I still would prefer if we don't promote `a, b, c` because it will lead to confusion for users which are not that familiar with YAML and are looking at other components/platforms.

Reply:
Great. The documentation pull request that I submitted uses option `d` for the example but I'm happy to document it with whichever form you think should be promoted.